src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/ShadowKmpPlugin.kt

```diff
@@ -48,4 +55,26 @@ public abstract class ShadowKmpPlugin : Plugin<Project> {
       }
     }
   }
+
+  private fun Project.addRunTask(target: KotlinJvmTarget) {
+//      if (KgpVersion.DEFAULT < KgpVersion.fromVersion("2.1.20")) return@configureEach
+
+    tasks.register(SHADOW_RUN_TASK_NAME, JavaExec::class.java) { task ->
+      task.description = "Runs this project as a JVM application using the shadow jar"
+      task.group = ApplicationPlugin.APPLICATION_GROUP
+
+      task.classpath = files(tasks.shadowJar)
+
+      target.binaries {
+        executable { dsl ->
```

Comment:
Any way to defer `binaries` and `executable`? I see `binariesDsl` is private, we can't use it...

https://github.com/JetBrains/kotlin/blob/0cdd9f9deb7ea0e89a3cfdd364adeb617a716e04/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/jvm/KotlinJvmTarget.kt#L98-L104

CC @antohaby

Reply:
CC @osipxd

Reply:
I've created an issue to address this:
[KT-77499](https://youtrack.jetbrains.com/issue/KT-77499) Provide a way to access and configure JVM binaries defined by users

Reply:
Thanks for this feedback!